python/mocpy/moc/moc.py

```diff
@@ -549,15 +549,26 @@ def from_fits_image(cls, hdu, max_norder, mask=None):
         frame = wcs.utils.wcs_to_celestial_frame(w)
         skycrd = SkyCoord(world, unit="deg", frame=frame)
 
-        # Compute the order based on the CDELT
-        # Should work even if the fits is defined using cd_ij or crota, see:
-        # https://docs.astropy.org/en/stable/api/astropy.wcs.Wcsprm.html#astropy.wcs.Wcsprm.get_cdelt
-        [c1, c2] = w.wcs.get_cdelt()
-
-        max_res_px = np.sqrt(c1 * c1 + c2 * c2) * np.pi / 180.0
-        max_depth_px = int(np.floor(np.log2(np.pi / (3 * max_res_px * max_res_px)) / 2))
-
-        max_norder = min(max_norder, max_depth_px)
+        # Compute the deepest HEALPix order containing at least one 1 pixel of the image
+        # We want the order so that area_hpx_cell >= area_img_pixel
+        # <=> 4pi / (12 * 2^(2*order)) in [steradians] >= area_img_pixel in [steradians]
+        [w_img_px, h_img_px] = hdu.data.shape
+        lonlat = w.wcs_pix2world([[0, 0], [w_img_px - 1, h_img_px - 1]], 0)
+
+        if np.isfinite(lonlat).all():
```

Comment:
what happens if it is not finite?

Reply:
or we can use the cdelt for computing the healpix order of a pixel image for the non finite case

Reply:
I prefer the first option